pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +161,46 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+// An empty string means that number of approvers is invalid
+func (e *Executor) validateApproverNum(ctx context.Context, approver string, minApproverNum int) bool {
+	if minApproverNum <= 1 {
+		e.LogPersister.Infof("Got approval from %q", approver)
+		return true
+	}
+	as, ok := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	if !ok {
+		e.LogPersister.Infof("Got approval from %q", approver)
+		e.LogPersister.Infof("Waiting for %d other approvers...", minApproverNum-1)
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approversKey, approver); err != nil {
+			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+		}
+		return false
+	}
+
+	for _, a := range strings.Split(as, ", ") {
+		if a == approver {
+			e.LogPersister.Infof("Approval from the same user (%s) will not be counted", approver)
+			return false
+		}
+	}
+	e.LogPersister.Infof("Got approval from %q", approver)
+
+	totalAs := fmt.Sprintf("%s, %s", as, approver)
+	if c := minApproverNum - len(strings.Split(totalAs, ", ")); c > 0 {
+		e.LogPersister.Infof("Waiting for %d other approvers...", c)
+		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approversKey, totalAs); err != nil {
+			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
+		}
+		return false
+	}
+	e.LogPersister.Info("Received all needed approvals")
```

Comment:
How do you think about this?
I think it is simpler and avoid using `Split` in many places.

```suggestion
        const delimiter = ", "
	as, _ := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
        var approvedUsers []string
        if as != "" {
           approvedUsers = strings.Split(as, delimiter)
        }
	for _, u := approvedUsers {
		if approver = u {
			e.LogPersister.Infof("Approval from the same user (%s) will not be counted", approver)
			return false
		}
	}
	
	approvedUsers = append(approvedUsers, approver)
	e.LogPersister.Infof("Got approval from %q", approver)

	if remain := minApproverNum - len(approvedUsers); remain > 0 {
		e.LogPersister.Infof("Waiting for %d other approvers...", remain)
		if err := e.MetadataStore.Stage(e.Stage.Id).Put(ctx, approversKey, strings.Join(approvedUsers, delimiter); err != nil {
			e.LogPersister.Errorf("Unable to save approver information to deployment, %v", err)
		}
		return false
	}
	e.LogPersister.Info("Received all needed approvals")
```